<?xml version="1.0"?>
<robot name="multipleshapes">

  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz=" 0 0 0.05"/>
    </visual>
  </link>

  <link name="platform">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.5"/>
      </geometry>
      <origin rpy="0 0 0" xyz=" 0 0 0.1"/>
    </visual>
  </link>
  
  <joint name="joint_base_link_to_platform" type="fixed">
    <parent link="base_link"/>
    <child link="platform"/>
  </joint>

  <link name="link1">
    <visual>
      <geometry>
        <cylinder length="1" radius="0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz=" 0 0 0.5"/>
    </visual>
  </link>

  <joint name="joint_platform_to_link1" type="revolute">
    <parent link="platform"/>
    <child link="link1"/>
    <origin rpy="0 0 0" xyz="0 0 0.2"/>
    <limit upper="1.5705" lower="-1.5705" effort="30" velocity="5.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  
    <link name="link2">
    <visual>
      <geometry>
        <cylinder length="1" radius="0.2"/>
      </geometry>
      <origin rpy="0 1.5705 0" xyz=" 0.3 0 0"/>
    </visual>
  </link>

  <joint name="joint_link2_to_link1" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="0 0 0" xyz="0 0 1"/>
    <limit upper="1.5705" lower="-1.5705" effort="30" velocity="5.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  
  <link name="link3">
    <visual>
      <geometry>
        <cylinder length="1.2" radius="0.2"/>
      </geometry>
      <origin rpy="0 1.5705 1.5705" xyz="0 0.4 0"/>
    </visual>
  </link>

  <joint name="joint_link3_to_link2" type="fixed">
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 0 0" xyz="0.6 0 0"/>
  </joint>
  
  <link name="link4">
    <visual>
      <geometry>
        <cylinder length="0.8" radius="0.2"/>
      </geometry>
      <origin rpy="0 -1.57 0" xyz="-0.2 0 0"/>
    </visual>
  </link>

  <joint name="joint_link4_to_link3" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin rpy="0 0 0" xyz="0 0.8 0"/>
    <limit upper="1.5705" lower="-1.5705" effort="30" velocity="5.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  
    <link name="link5">
    <visual>
      <geometry>
        <cylinder length="1" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.1 0 -0.5"/>
    </visual>
  </link>

  <joint name="joint_link5_to_link4" type="fixed">
    <parent link="link4"/>
    <child link="link5"/>
    <origin rpy="0 0 0" xyz="-0.6 0 0"/>
  </joint>

</robot>
